### ISSUE TITLE
Modified Loss Function

### ISSUE THREAD

**Issue by alifanta** [jakeret/tf_unet#233]
Hi, thank you for your U-net implementation.
I am using your code to segment the aortic lumen from CTA images. I've tried both cross entropy and dice losses, with bad results. I think that the reason is that the classes are highly unbalanced. 
I have implemented my own loss function to separately handle the class errors in this way:

elif cost_name == "my_cost":
                eps = 1e-5
                # Define error in aorta identification
                prediction = pixel_wise_softmax(logits)
                prediction = prediction[..., 1]
                gt_map = self.y[...,1]                
                AortaError = tf.reduce_sum(gt_map - tf.math.multiply(prediction,gt_map)) / (tf.reduce_sum(gt_map) + eps)
                # Define background error
                BGError = tf.reduce_sum(tf.math.multiply(prediction, (1-gt_map))) / (tf.reduce_sum(1-gt_map) + eps)                
                # Use different weights                
                alpha = 0.5                 
                loss = (alpha*AortaError + (1 - alpha)*BGError)

I have used layers=4, features_root=64 and 400 iterations with batch_size = 6. 
After the first epoch I get this result: 
![epoch_0](https://user-images.githubusercontent.com/44367682/49869493-d1f18b80-fe10-11e8-8c65-52803fb1a9a6.jpg)
I think the network is segmenting the aorta without considering the contextual information but only the graylevels. Do you think the problem is in  my loss function?

Thanks for your help, 

Alice

**Comment by daboe01** [comment]
@alifanta  just curious, does this loss function work for you?
 https://github.com/xuuuuuuchen/Active-Contour-Loss/blob/master/Active-Contour-Loss.py
